Branch-SDK/Branch-SDK/ContentDiscoverer.m

```diff
@@ -0,0 +1,289 @@
+        }
+    }
+}
+
+
+- (NSString *)getViewText:(NSString *)viewId forController:(UIViewController *)viewController {
+    NSString *viewTxt = @"";
+    if (viewController) {
+        UIView *rootView = [viewController view];
+        NSArray *viewIDsplitArray = [viewId componentsSeparatedByString:@":"];
+        if (viewIDsplitArray.count > 0) {
+            viewId = [[viewId componentsSeparatedByString:@":"] objectAtIndex:1];
+        }
+        NSArray *viewIds = [viewId componentsSeparatedByString:@"-"];
+        BOOL foundView = YES;
+        for (NSString *subViewIdStr in viewIds) {
+            NSInteger subviewId = [subViewIdStr intValue];
+            if ([rootView isKindOfClass:UITableView.class] || [rootView isKindOfClass:UICollectionView.class]) {
+                NSArray *cells = [rootView performSelector:@selector(visibleCells) withObject:nil];
+                if (cells.count > subviewId) {
+                    rootView = [cells objectAtIndex:subviewId];
+                } else {
+                    foundView = NO;
+                    break;
+                }
+            } else {
+                if ([rootView subviews].count > subviewId) {
+                    rootView = [[rootView subviews] objectAtIndex:subviewId];
+                } else {
+                    foundView = NO;
+                    break;
+                }
+            }
+        }
+        if (foundView) {
+            NSString *contentVal = [self getContentText:rootView];
+            if (contentVal) {
+                viewTxt = contentVal;
+            }
+        }
+    }
+    return viewTxt;
+}
+
+- (UIViewController *)getActiveViewController {
+    UIViewController *rootViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
+    return [self getActiveViewController:rootViewController];
+    
+}
+
+- (UIViewController *)getActiveViewController:(UIViewController *)rootViewController {
+    UIViewController *activeController;
+    if ([rootViewController isKindOfClass:[UINavigationController class]]) {
+        activeController = ((UINavigationController *)rootViewController).topViewController;
+    } else if ([rootViewController isKindOfClass:[UITabBarController class]]) {
+        activeController = ((UITabBarController *)rootViewController).selectedViewController;
+    } else {
+        activeController = rootViewController;
+    }
+    return activeController;
+}
+
+- (void)addFormatedContentData:(NSMutableArray *)contentDataArray withText:(NSString *)contentData clearText:(BOOL)isClearText {
+    if (contentData && contentData.length > _cdManifest.maxTextLen) {
+        contentData = [contentData substringToIndex:_cdManifest.maxTextLen];
+    }
+    if (!isClearText) {
+        contentData = [self hashContent:contentData];
+    }
+    [contentDataArray addObject:contentData];
+}
+
+- (NSString*)hashContent:(NSString *)content {
```

Comment:
space before the `*` in the return type
might be a good candidate to add to the encoding utils class